src/dag-service.js

```diff
@@ -23,10 +25,27 @@ function DAGService (blockService) {
   // this.addRecursive
 
   // get retrieves a DAGNode, using the Block Service
-  this.get = (multihash, callback) => {
-    if (!multihash) { return callback(new Error('Invalid Key')) }
+  this.get = function (multihash, callback) {
+    const isMhash = isIPFS.multihash(multihash)
+    const isPath = isIPFS.path(multihash)
+
+    if (!isMhash && !isPath) {
+      return callback(new Error('Invalid Key'))
+    }
+
+    if (isMhash) {
+      this.getWith(multihash, callback)
+    }
+
+    if (isPath) {
+      var ipfsKey = new Buffer(base58.decode(multihash.replace('/ipfs/', '')))
+      this.getWith(ipfsKey, callback)
+    }
+  }
 
-    this.bs.getBlock(multihash, (err, block) => {
+  this.getWith = function (key, callback) {
+    const formatted = typeof key === 'string' ? new Buffer(base58.decode(key)) : key
```

Comment:
seems that there is already the formatting happening on line 37 and 41, is this necessary?